src/style.css

```diff
@@ -0,0 +1,118 @@
+* {
+  box-sizing: border-box;
+}
+
+body {
+  margin: 0;
+  font-family: "Roboto", sans-serif;
+  color: #000;
+}
+
+.product {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  padding: 30px 14px 14px;
+  width: 200px;
+  background: #fff;
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+}
+
+.product__image {
+  height: 134px;
+  width: 160px;
+  margin-bottom: 45px;
+}
+
+.product__name {
+  font-weight: 500;
+  font-size: 12px;
+  line-height: 15px;
+}
+
+.product__code {
+  align-self: start;
+  font-size: 11px;
+  line-height: 13px;
+  margin-top: 5px;
+}
+
+.product__rate {
+  display: flex;
+  justify-content: space-between;
+  width: 100%;
+  margin-top: 15px;
+}
+
+.product__rate_stars {
+  display: flex;
+}
+
+.stars__star {
+  background-image: url(images/star.svg);
+  background-repeat: no-repeat;
+  background-position: center;
+  width: 15px;
+  height: 15px;
+  margin-right: 4px;
+}
+
+.product__rate_stars .stars__star:nth-child(-n+4) {
+  background-image: url(images/star-active.svg);
+}
+
+.product__rate_reviews {
+  font-size: 11px;
+  line-height: 13px;
+  width: 100%;
+  text-align: end;
+}
+
+.product__price {
+  display: flex;
+  justify-content: space-between;
+  align-items: baseline;
+  width: 100%;
+  margin-top: 20px;
+}
+
+.product__price_title {
+  font-size: 11px;
+  line-height: 13px;
+}
+
+.product__price_amount {
+  font-weight: bold;
+  font-size: 16px;
+  line-height: 19px;
+}
+
+.product__button {
+  margin-top: 20px;
+  height: 40px;
+  width: 100%;
+}
+
+.product__button_link {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+  height: 100%;
+  background: #00acdc;
+  border-radius: 5px;
+  color: #fff;
+  font-weight: bold;
+  font-size: 14px;
+  line-height: 16px;
+  text-align: center;
+  text-transform: uppercase;
+  text-decoration: none;
+}
+
+.product__button_link:hover {
+  background: #fff;
+  border: 1px solid #00acdc;
```

Comment:
at this moment when you hovered on your link - border appears and adds 1px, because of box-sizing: border-box; width and height of your button is recalculating, and text could change his position, so its better if you write border property inside the button.